lib/ansible/modules/cloud/misc/qemu_img.py

```diff
@@ -0,0 +1,221 @@
+    shrink = module.boolean(module.params['shrink'])
+
+    qemu_img = module.get_bin_path('qemu-img', True)
+
+    if state == 'present':
+        if not size:
+            module.fail_json(msg="No size defined, creating a disk image requires a size")
+        if not os.path.exists(dest):
+            if not module.check_mode:
+                if not img_options:
+                    module.run_command('%s create -f %s "%s" %s' % (qemu_img, img_format, dest, size), check_rc=True)
+                else:
+                    module.run_command('%s create -f %s -o "%s" "%s" %s' % (qemu_img, img_format, img_options, dest, size), check_rc=True)
+            result['changed'] = True
+        else:
+            size_unit = size[-1:]
+            if size_unit not in ['T','G','M','k','b','']:
+                module.fail_json(msg="No valid size unit specified")
+            else:
+                if(size_unit == 'T'):
+                    unitmultiplier = 1024*1024*1024*1024
+                elif(size_unit == 'G'):
+                    unitmultiplier = 1024*1024*1024
+                elif(size_unit == 'M'):
+                    unitmultiplier = 1024*1024
+                elif(size_unit == 'k'):
+                    unitmultiplier = 1024
+                else:
+                    unitmultiplier = 1
+            size = size[:-1]
+            size_increase = False
+            size_decrease = False
+            if(size[:1] == '+'):
+                size_increase = True
+                size = size[1:]
+            elif(size[:1] == '-'):
+                size_descrease = True
+                size = size[1:]
+            size = int(float(size)*float(unitmultiplier))
+            rc, stdout, _ = module.run_command('%s info "%s"' % (qemu_img, dest), check_rc=True)
+            current_size = None
+            for line in stdout.splitlines():
+                if 'virtual size' in line:
+                    ### virtual size: 5.0M (5242880 bytes)
+                    current_size = int(line.split('(')[1].split()[0])
+            if not current_size:
+                module.fail_json(msg='Unable to read virtual disk size of %s' % (dest))
+            if(grow):
+                newsize = size - current_size
+                if(newsize > 0):
+                    if not module.check_mode:
+                        module.run_command('%s resize "%s" %s' % (qemu_img, dest, size), check_rc=True)
+                    result['changed'] = True
+            if(shrink):
+                if(img_format == 'qcow2'):
+                    module.fail_json(msg="qemu-img does not support shrinking qcow2 images")
+                if (size[:1] == '-'):
+                    newsize = current_size - size
+                else:
+                    newsize = current_size - size
+                if(newsize > 0):
+                    if not module.check_mode:
+                        module.run_command('%s resize "%s" %s' % (qemu_img, dest, size), check_rc=True)
```

Comment:
The final command should probably return its stdout, stderr and rc back to the playbook.